src/engraving/rw/compat/compatutils.cpp

```diff
@@ -406,3 +509,63 @@ void CompatUtils::resetRestVerticalOffset(MasterScore* masterScore)
         }
     }
 }
+
+void CompatUtils::resetArticulationOffsets(MasterScore* masterScore)
+{
+    for (Score* score : masterScore->scoreList()) {
+        for (Measure* measure = score->firstMeasure(); measure; measure = measure->nextMeasure()) {
+            for (Segment& segment : measure->segments()) {
+                if (!segment.isChordRestType()) {
+                    continue;
+                }
+                for (EngravingItem* item : segment.elist()) {
+                    if (!item || !item->isChord()) {
+                        continue;
+                    }
+                    Chord* chord = toChord(item);
+                    for (Articulation* artic : chord->articulations()) {
+                        if (!artic) {
+                            continue;
+                        }
+                        // vertical offset reset in all situations
+                        if (artic->offset().y() != 0) {
+                            PointF newOffset = PointF(artic->offset().x(), 0.0);
+                            artic->setProperty(Pid::OFFSET, newOffset);
+                        }
+                        // horizontal offset reset if on stem-side
+                        if (artic->up() == chord->up() && artic->offset().x() != 0) {
```

Comment:
Is this data reliable? I.e. are we sure that layout has already been computed at this point?

Reply:
That's a very good point. I've tested it and the chord direction at this point is indeed not reliable. For the time being I have pushed the version that resets all offsets